src/reader/mod.rs

```diff
@@ -458,8 +459,36 @@ pub struct Reader<R> {
     parser: Parser,
 }
 
-/// Builder methods
+/// Methods specialized to reading from a slice
+impl<'a> Reader<&'a [u8]> {
+    /// Creates an XML reader from a string slice.
+    pub fn from_str(s: &'a str) -> Self {
+        // Rust strings are guaranteed to be UTF-8, so lock the encoding
+        #[cfg(feature = "encoding")]
+        {
+            let mut parser = Parser::default();
+            parser.encoding = EncodingRef::Explicit(UTF_8);
+            Self {
+                reader: s.as_bytes(),
+                parser: parser,
+            }
+        }
+
+        #[cfg(not(feature = "encoding"))]
+        {
+            Self {
+                reader: s.as_bytes(),
+                parser: Parser::default(),
+            }
+        }
+    }
+}
+
+/// Public implementation-independent functionality
 impl<R> Reader<R> {
+    /// Configuration methods
```

Comment:
I'm not sure that doc comment is allowed here or has some meaning
```suggestion
    // Configuration methods
```